mpas_analysis/ocean/climatology_map_waves.py

```diff
@@ -0,0 +1,526 @@
+           A list of seasons (keys in ``constants.monthDictionary``) over
+           which the climatology should be computed.
+
+        fileName : str
+            The name of the observation file
+
+        outFilePrefix : str
+            The prefix in front of output files and mapping files, typically
+            the name of the field being remapped
+
+        variableList : list
+            List of observational variables to remap
+
+        comparisonGridNames : list of {'latlon', 'antarctic', 'arctic'},
+            optional
+            The name(s) of the comparison grid to use for remapping.
+
+        subtaskName : str, optional
+            The name of the subtask
+        '''
+        # Authors
+        # -------
+        # Steven Brus
+
+        self.variableList = variableList
+
+        # call the constructor from the base class (AnalysisTask)
+        super().__init__(parentTask, seasons, fileName, outFilePrefix,
+                         comparisonGridNames=['latlon'],
+                         subtaskName=subtaskName)
+        # }}}
+
+    def get_observation_descriptor(self, fileName):  # {{{
+        '''
+        get a MeshDescriptor for the observation grid
+
+        Parameters
+        ----------
+        fileName : str
+            observation file name describing the source grid
+
+        Returns
+        -------
+        obsDescriptor : ``MeshDescriptor``
+            The descriptor for the observation grid
+        '''
+        # Authors
+        # -------
+        # Steven Brus
+
+        # create a descriptor of the observation grid using the lat/lon
+        # coordinates
+        obsDescriptor = LatLonGridDescriptor.read(fileName=fileName,
+                                                  latVarName='lat',
+                                                  lonVarName='lon')
+        return obsDescriptor  # }}}
+
+    def build_observational_dataset(self, fileName):  # {{{
+        '''
+        read in the data sets for observations, and possibly rename some
+        variables and dimensions
+
+        Parameters
+        ----------
+        fileName : str
+            observation file name
+
+        Returns
+        -------
+        dsObs : ``xarray.Dataset``
+            The observational dataset
+        '''
+        # Authors
+        # -------
+        # Steven Brus
+
+        sectionName = self.taskName
+        climStartYear = self.config.getint(sectionName, 'SscciObsStartYear')
+        climEndYear = self.config.getint(sectionName, 'SscciObsEndYear')
+        timeStart = datetime.datetime(year=climStartYear, month=1, day=1)
+        timeEnd = datetime.datetime(year=climEndYear, month=12, day=31)
+
+        dsObs = xr.open_dataset(fileName)
+        dsObs = dsObs.rename({'time': 'Time'})
+        dsObs = dsObs.sel(Time=slice(timeStart, timeEnd))
+        dsObs.coords['month'] = dsObs['Time.month']
+        dsObs.coords['year'] = dsObs['Time.year']
+
+        dsObs = dsObs[self.variableList]
+
```

Comment:
Here again something like:
```suggestion

        degrees = 'degree' in dsObs.lon.units
        dsObs = add_periodic_lon(ds=dsObs, lonDim='lon', degrees=degrees)
```